lib/runtime/wazero/imports.go

```diff
@@ -2198,8 +2254,33 @@ func ext_storage_root_version_1(ctx context.Context, m api.Module) uint64 {
 }
 
 func ext_storage_root_version_2(ctx context.Context, m api.Module, version uint32) uint64 { //skipcq: RVV-B0012
-	// TODO: update to use state trie version 1 (#2418)
-	return ext_storage_root_version_1(ctx, m)
+	rtCtx := ctx.Value(runtimeContextKey).(*runtime.Context)
+	if rtCtx == nil {
+		panic("nil runtime context")
+	}
+	storage := rtCtx.Storage
+
+	stateVersionBytes, _ := m.Memory().Read(version, 4)
+	_, err := trie.ParseVersion(binary.LittleEndian.Uint32(stateVersionBytes))
+	if err != nil {
+		logger.Errorf("failed parsing state version: %s", err)
+		return mustWrite(m, rtCtx.Allocator, emptyByteVectorEncoded)
+	}
+
+	root, err := storage.Root()
+	if err != nil {
+		logger.Errorf("failed to get storage root: %s", err)
+		panic(err)
+	}
+
+	logger.Debugf("root hash is: %s", root)
+
+	rootSpan, err := write(m, rtCtx.Allocator, root[:])
+	if err != nil {
+		logger.Errorf("failed to allocate: %s", err)
+		panic(err)
+	}
+	return rootSpan
```

Comment:
from the comment found on substrate discussion ([link](https://github.com/paritytech/substrate/discussions/11607#discussioncomment-3102924)) I think a call to this host function is supposed to trigger a migration on the trie.   So all values > 32 bytes are supposed to be hashed from this point forward if the parsed version is v1.

Reply:
I don't think this function should be responsible for that migration, they created a [pallet](https://github.com/paritytech/substrate/pull/10073) for that purpose. 

As far as I understand this function only calculates the root for the current state using 2 strategies depending on version parameter (v1 or v2). It does not change the values in the trie.